ui/pages/confirmations/components/confirm/info/typed-sign/permit-simulation/permit-simulation.tsx

```diff
@@ -0,0 +1,81 @@
+import React, { useEffect, useState } from 'react';
+import { useSelector } from 'react-redux';
+import { NameType } from '@metamask/name-controller';
+
+import { Numeric } from '../../../../../../../../shared/modules/Numeric';
+import Name from '../../../../../../../components/app/name/name';
+import {
+  ConfirmInfoRow,
+  ConfirmInfoRowText,
+} from '../../../../../../../components/app/confirm/info/row';
+import { useI18nContext } from '../../../../../../../hooks/useI18nContext';
+import { currentConfirmationSelector } from '../../../../../../../selectors';
+import { Box, Text } from '../../../../../../../components/component-library';
+import {
+  BackgroundColor,
+  BorderRadius,
+  Display,
+  TextAlign,
+} from '../../../../../../../helpers/constants/design-system';
+import { SignatureRequestType } from '../../../../../types/confirm';
+import useTokenExchangeRate from '../../../../../../../components/app/currency-input/hooks/useTokenExchangeRate';
+import { IndividualFiatDisplay } from '../../../../simulation-details/fiat-display';
+
+const PermitSimulation: React.FC = () => {
+  const t = useI18nContext();
+  const currentConfirmation = useSelector(
+    currentConfirmationSelector,
+  ) as SignatureRequestType;
+  const [fiatValue, setFiatValue] = useState<number>();
+
+  const {
+    domain: { verifyingContract },
+    message: { value },
+  } = JSON.parse(currentConfirmation?.msgParams?.data as string);
+
+  const exchangeRate = useTokenExchangeRate(verifyingContract);
+
+  useEffect(() => {
```

Comment:
Since we're not using any asynchronous logic within this `useEffect` and not exposing the `setFiatValue` outside of this component, do we need the `useState` and `useEffect` at all?

We could use a `useMemo` if the logic had a performance cost, or we were generating an object, but since we're generating a standard `number` with a basic multiplication, could we remove both for simplicity?

Reply:
Putting `fiatValue` in state ensures re-rendering when it changes.

Reply:
It should re-render regardless whenever the properties in the JSX changes, including `fiatValue`.

Reply:
re-rendering is done in react for change in state and props, it does not keep track of value used in jsx

but I guess change in `exchangeRate` and `currentConfirmation` will cause re-render too as its returned by hook

Reply:
I updated the PR.